test/TesterInternal/RemindersTest/ReminderTableTestsBase.cs

```diff
@@ -113,7 +119,12 @@ protected virtual string GetAdoInvariant()
             await Task.WhenAll(Enumerable.Range(1, iterations).Select(async i =>
             {
                 GrainReference grainRef = MakeTestGrainReference();
-                await remindersTable.UpsertRow(CreateReminder(grainRef, i.ToString()));
+
+                await RetryHelper.RetryOnExceptionAsync<Task>(10, RetryOperation.Sigmoid, async () =>
+                {
+                    await remindersTable.UpsertRow(CreateReminder(grainRef, i.ToString()));
+                    return Task.CompletedTask;
```

Comment:
This seems a little confusing to me: is the `operation` parameter to `RetryOnExceptionAsync` being inferred as `Func<Task<Task>>`? I think an overload is needed which allows `Func<Task>` instead. The inner task (`Task.CompletedTask`) is never awaited, from what I can tell. That's fine in this case, because it's a synchronously-completed, fixed value, but it could lead to a subtle bug in future.

Could instead add `.Unwrap()` to the `RetryOnExceptionAsync()` call, I think.

Reply:
@ReubenBond Or call `await await`, might be entirely possible I just wrote noisy syntax. This `RetryHelper` is about five minutes of work and more like an internal, tempory resiliency measure in the CI (and slow machines) It's worth thinking if a robust one is needed and should be created or something else, such as Polly, be used. Either way, it occurred to me a set of precanned, parametrizable functions such as sigmoid, linear, exponential backoff etc. could be useful, together with diagrams.